File: 0.36/emulator/remaining_mame_src/sound/tiaintf.c
/***************************************************************************

   tiaintf.c

   Interface the routines in tiasound.c to MESS

***************************************************************************/

#include "driver.h"
#include "sound/tiasound.h"
#include "sound/tiaintf.h"

#define MIN_SLICE 10    /* minimum update step */

#define BUFFER_LEN 8192

static unsigned int sample_pos;

static int channel;

static const struct TIAinterface *intf;
static INT16 *buffer;

int tia_sh_start(const struct MachineSound *msound)
{
//	int i, res;

	intf = msound->sound_interface;

	if (Machine->sample_rate == 0) return 0;
	sample_pos = 0;

	channel = mixer_allocate_channel(intf->volume);

	if ((buffer = malloc(sizeof(INT16) * BUFFER_LEN)) == 0)
		return 1;
	memset(buffer,0,sizeof(INT16) * BUFFER_LEN);

	Tia_sound_init (intf->clock, Machine->sample_rate);
	return 0;
}


void tia_sh_stop (void)
{
	/* Nothing to do here */
}

void tia_sh_update (void)
{
	int buflen;

	if (Machine->sample_rate == 0) return;

	buflen = mixer_samples_this_frame();
	if (sample_pos < buflen)
		Tia_process (buffer + sample_pos, buflen - sample_pos);
	sample_pos = 0;

	mixer_play_streamed_sample_16(channel,buffer,2*buflen,Machine->sample_rate);
}

File: 0.37b7/converter/java_code/sound/tiaintf.java
/*
 * ported to v0.37b7
 * using automatic conversion tool v0.01
 */ 
package sound;

public class tiaintf
{
	
	static int channel = -1;
	static const struct TIAinterface *intf;
	
	public static ShStartPtr tia_sh_start = new ShStartPtr() { public int handler(MachineSound msound) 
	{
	    intf = msound.sound_interface;
	
	    if (Machine.sample_rate == 0)
	        return 0;
	
		channel = stream_init("TIA", intf.volume, Machine.sample_rate, 0, tia_process);
		if (channel == -1)
	        return 1;
	
		tia_sound_init(intf.clock, Machine.sample_rate, intf.gain);
	
	    return 0;
	} };
	
	
	public static ShStopPtr tia_sh_stop = new ShStopPtr() { public void handler() 
	{
	    /* Nothing to do here */
	} };
	
	public static ShUpdatePtr tia_sh_update = new ShUpdatePtr() { public void handler() 
	{
		stream_update(channel, 0);
	} };
	
	
}
